src/exchange.c

```diff
@@ -472,6 +493,11 @@ enum wps_type process_wps_message(const void *data, size_t data_size)
                                         case MESSAGE_TYPE:
                                                 type = (uint8_t) element_data[0];
                                                 break;
+					case CONFIGURATION_ERROR:
+						/* Check element_data length */
+						if (element.length == 2)
+							set_nack_reason(htons(*( (uint16_t*) element_data)));
```

Comment:
i guess WPA_GET_BE16(element_data) would be more readable (and no potential issues with unaligned access).
also maybe we should initialize the nack reason to something like -1 before entering this block, so we dont mistakenly interpret a previous nack as one.
does a nack always cause the CONFIGURATION_ERROR condition here to get triggered ?

Reply:
> does a nack always cause the CONFIGURATION_ERROR condition here to get triggered ?

Yes, the CONFIGURATION_ERROR and MESSAGE_TYPE (NACK(0x0E)) are in the same packet, and its are processed.